### DISCUSSION TITLE
Graphical Glitches on Linux

### DISCUSSION THREAD

**Discussion by zen1337th** [ppy/osu#28671] [Q&A]
### Type

Game behaviour

### Bug description

Scrolling through the menus results them to disappear using only the OpenGL (+Experimental) renderer. (Vulkan works just fine.)
Video attached below.

Other menus besides the osu!direct menu are not effected as much as the osu!direct menu is.

My graphics card is an NVIDIA GeForce GTX 1080, using the NVIDIA Driver Version 550.90.07 on Arch Linux.

Issue is reproduceable on my Laptop using an i7-8650U, also using the same Arch Linux distro.

So far I had no issue during gameplay.

### Screenshots or videos

https://github.com/ppy/osu/assets/104238285/df6e7504-ac12-463c-96a2-72313cec1137

### Version

2024.625.2

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16036998/compressed-logs.zip)

**Comment by smoogipoo** [comment]
Three things here...

First is that this is the in-house GL renderer rather than the experimental one (since that one's been turned off). You could try using the experimental renderer manually via environment variables:
```
OSU_GRAPHICS_RENDERER=deferred
OSU_GRAPHICS_SURFACE=opengl
```
Would be curious to know if that behaves any better. Vulkan is still unstable so I hesitate recommending to use it.

Second is I'm not sure when we're ever going to have the bandwidth to fix this. At the end of the day it's going to be a driver thing - so perhaps updating your Mesa drivers/etc will resolve, but it's difficult enough to just get the affected hardware much less testing on Linux.

Third, I'm currently working on WebGPU support which may help in the future.

**Reply by zen1337th** [reply]
The issue still keeps being present as it was with the experimental renderer you gave me.

Also I have figured out recently that this might be an issue with Wayland because after switching to X11 this issue has completely disappeared until I switched back to Wayland and then reappeared again. (Speaking for both devices that I used for testing)

Like I shortly mentioned; I also tried using Vulkan and had no issues with it at all. Now I also tried it on both Wayland and X11 which has worked as fine as OpenGL did after switching to X11. At least that was the case for me.

Since this hasn't affected the gameplay for me, I would simply just recommend switching to X11 which also helps to fix some other graphical issues in other games in general.
